nativelib/src/main/resources/atomic.cpp

```diff
@@ -0,0 +1,537 @@
+#include <atomic>
+#include <stdint.h>
+#include <stdlib.h>
+
+using namespace std;
+
+extern "C" {
+
+	/**
+     * Init
+     * */
+
+     // byte
+     void init_byte(int8_t* atm, int8_t init_value) {
```

Comment:
Since the functions are there to "create" the atomic object I thought it'd be maybe clearer to keep the parameter as is to mark the fact that it is a creation of an atomic.